ts/src/krakenfutures.ts

```diff
@@ -1591,6 +1591,86 @@ export default class krakenfutures extends Exchange {
         return this.safeBalance (result);
     }
 
+    async fetchFundingRate (symbol, params = {}) {
+        const fundingRates = await this.fetchFundingRates ([ symbol ], params);
+        const fundingRate = this.safeValue (fundingRates, 0);
+        if (fundingRate === undefined) {
+            throw new BadSymbol (this.id + ' does not have market symbol ' + symbol);
+        }
+        return fundingRate;
+    }
+
+    async fetchFundingRates (symbols: string[] = undefined, params = {}) {
+        await this.loadMarkets ();
+        const marketIds = this.marketIds (symbols);
+        const response = await this.publicGetTickers (params);
+        const tickers = this.safeValue (response, 'tickers');
+        const fundingRates = [];
+        for (let i = 0; i < tickers.length; i++) {
+            const entry = tickers[i];
+            const entry_symbol = this.safeValue (entry, 'symbol');
+            if (marketIds !== undefined) {
+                if (!this.inArray (entry_symbol, marketIds)) {
+                    continue;
+                }
+            }
+            const market = this.safeMarket (entry_symbol);
+            const parsed = this.parseFundingRate (entry, market);
+            fundingRates.push (parsed);
```

Comment:
@youknowone one more thing, I think this method should return an object indexed by the symbol and not a list

Reply:
Ah, I finally realized `filter_by_array` can return either array or object.
I will add index_by call to the result.

Reply:
@youknowone thanks!